tests/OmniSharp.Tests/DiagnosticsFacts.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OmniSharp.Models;
+using Xunit;
+
+namespace OmniSharp.Tests
+{
+    public class DiagnosticsFacts
+    {
+        [Fact]
+        public async Task CodeCheckSpecifiedFileOnly()
+        {
+            var workspace = TestHelpers.CreateSimpleWorkspace(new Dictionary<string, string>
+            {
+                { "a.cs", "class C { int n = true; }" }
+            });
+
+            var controller = new OmnisharpController(workspace, null);
+            var quickFixes = await controller.CodeCheck(new Request() { FileName = "a.cs" });
+
+            Assert.Equal(2, quickFixes.QuickFixes.Count());
```

Comment:
That's a good question ;-) There are two errors because we have two projects, but then we have logic in place to de-dup error messages. I think it's a regression which we never saw due to be lack of a test (https://github.com/OmniSharp/omnisharp-roslyn/blame/master/src/OmniSharp/Models/QuickFix.cs#L36) 

I wonder if @Yantrio  knows if equality of a quick fix really requires the list of projects or we should use a special comparator to de-dup quick fixes.